compiler/daemon/daemon-tests/test/org/jetbrains/kotlin/daemon/report/BuildReportICReporterTest.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.daemon.report
+
+import org.jetbrains.kotlin.build.report.debug
+import org.jetbrains.kotlin.build.report.info
+import org.jetbrains.kotlin.build.report.warn
+import org.jetbrains.kotlin.daemon.testfixtures.FakeCompilationResults
+import org.junit.Test
+import java.io.File
+import kotlin.test.assertEquals
+
+class BuildReportICReporterTest {
+
+    @Test
+    fun testVerboseMode() {
+        doTest(verbose = true)
+    }
+
+    @Test
+    fun testNonVerboseMode() {
+        doTest(verbose = false)
+    }
+
+    private fun doTest(verbose: Boolean) {
```

Comment:
Done - I've amended the top commit.

I've also removed the `BuildReporter*Async` tests that I just added in this PR as they turned out to be flaky, probably because in `RemoteICReporter.kt`, there are usages of `GlobalScope.async` without waiting for the work to finish:

```
    override fun flush() {
        GlobalScope.async {
            compilationResults?.add(CompilationResultCategory.BUILD_REPORT_LINES.code, icLogLines)
        }
    }
```    

Also rebased on master and fixed some trivial conflicts.